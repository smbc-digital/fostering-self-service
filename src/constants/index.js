export const Applicant = {
    FirstApplicant: 'firstApplicant',
    SecondApplicant: 'secondApplicant',
}

export const SECOND_APPLICANT_ROUTE = 'second-applicant'