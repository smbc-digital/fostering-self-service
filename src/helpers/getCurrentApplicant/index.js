import { Applicant, SECOND_APPLICANT_ROUTE } from 'constants'

const getCurrentApplicant = match => {
    if (match.params && match.params[0] === SECOND_APPLICANT_ROUTE) {
        return Applicant.SecondApplicant
    }

    return Applicant.FirstApplicant
}

export default getCurrentApplicant