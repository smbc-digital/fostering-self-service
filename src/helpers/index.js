import fetchWithTimeout from './fetchWithTimeout'
import { getPageRoute } from './pagehelper'
import getCurrentApplicant from './getCurrentApplicant'
import { updateFormStatus, updateForm, FormName, TaskStatus } from './updateForm'

export {
    fetchWithTimeout,
    getPageRoute,
    getCurrentApplicant,
    updateFormStatus,
    updateForm,
    FormName,
    TaskStatus
}