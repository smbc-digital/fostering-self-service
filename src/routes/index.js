import { StageName } from 'helpers'
import Start from 'components/Pages/Start'
import { KnownByAnotherName, MoreAboutYou } from 'components/Pages/HomeVisit/1-TellUsAboutYou'
import { AreYouEmployed, EmploymentDetails } from 'components/Pages/HomeVisit/2-YourEmploymentDetails'
import LanguagesSpokenInYourHome from 'components/Pages/HomeVisit/3-LanguagesSpokenInYourHome'
import { AreYouMarried, MarriageDate, MovedInTogetherDate } from 'components/Pages/HomeVisit/4-PartnershipStatus'
import { HaveYouPreviouslyApplied } from 'components/Pages/HomeVisit/5-YourFosteringHistory'
import YourHealth from 'components/Pages/HomeVisit/6-YourHealth'
import { AboutYourInterest, TypesOfFostering } from 'components/Pages/HomeVisit/7-InterestInFostering'
import { AnyUnderSixteens, AboutUnderSixteens, AnyOverSixteens, AboutOverSixteens } from 'components/Pages/HomeVisit/8-ChildrenAwayFromHome'
import { AnyPeopleInYourHousehold, PeopleInYourHousehold, DoYouHaveAnyPets } from 'components/Pages/HomeVisit/10-YourHousehold'
import { YourGpDetails } from 'components/Pages/Application'
import { FamilyReference, FirstPersonalReference, SecondPersonalReference } from 'components/Pages/Application/3-YourReferences'
import { DoYouKnowCouncillors } from 'components/Pages/Application/4-CouncillorsOrEmployees'

export const START_PAGE = '/fostering/your-fostering-journey'
export const KNOWN_BY_ANOTHER_NAME = '/fostering/known-by-another-name'
export const MORE_ABOUT_YOU = '/fostering/more-about-you'
export const ARE_YOU_EMPLOYED = '/fostering/are-you-employed'
export const EMPLOYMENT_DETAILS = '/fostering/employment-details'
export const LANGUAGES_SPOKEN_IN_YOUR_HOME = '/fostering/languages-spoken-in-your-home'
export const PARTNERSHIP_STATUS = '/fostering/your-partnership-status'
export const DATE_OF_MARRIAGE_OR_PARTNERSHIP = '/fostering/date-of-marriage-or-partnership'
export const DATE_OF_MOVE_IN_TOGETHER = '/fostering/date-of-move-in'
export const FOSTERING_HISTORY = '/fostering/your-fostering-history'
export const ABOUT_YOUR_HEALTH = '/fostering/about-your-health'
export const YOUR_INTEREST_IN_FOSTERING = '/fostering/your-interest-in-fostering'
export const TYPES_OF_FOSTERING = '/fostering/types-of-fostering'
export const YOUR_HOUSEHOLD = '/fostering/your-household'
export const WHO_LIVES_IN_YOUR_HOME = '/fostering/who-lives-in-your-home'
export const DO_YOU_HAVE_ANY_PETS = '/fostering/do-you-have-any-pets'
export const CHILDREN_UNDER_SIXTEEN_LIVING_AWAY = '/fostering/children-under-sixteen-living-away'
export const ABOUT_CHILDREN_UNDER_SIXTEEN_LIVING_AWAY = '/fostering/about-children-under-sixteen-living-away'
export const CHILDREN_OVER_SIXTEEN_LIVING_AWAY = '/fostering/children-over-sixteen-living-away'
export const ABOUT_CHILDREN_OVER_SIXTEEN_LIVING_AWAY = '/fostering/about-children-over-sixteen-living-away'
export const ABOUT_YOUR_GP = '/fostering/about-your-gp'
export const FAMILY_REFERENCE = '/fostering/family-reference'
export const FIRST_PERSONAL_REFERENCE = '/fostering/first-personal-reference'
export const SECOND_PERSONAL_REFERENCE = '/fostering/second-personal-reference'
export const RELATIONSHIP_TO_COUNCIL_EMPLOYEES = '/fostering/relationship-to-council-employees'
export const ABOUT_RELATIONSHIP_TO_COUNCIL_EMPLOYEES = '/fostering/about-relationship-to-council-employees'

export const RouterConfig = [
    {
        path: START_PAGE,
        component: Start
    },
    {
        path: KNOWN_BY_ANOTHER_NAME,
        component: KnownByAnotherName,
        stage: StageName.HomeVisit,
        bothApplicants: true
    },
    {
        path: MORE_ABOUT_YOU,
        component: MoreAboutYou,
        stage: StageName.HomeVisit,
        bothApplicants: true
    },
    {
        path: ARE_YOU_EMPLOYED,
        component: AreYouEmployed,
        stage: StageName.HomeVisit,
        bothApplicants: true
    },
    {
        path: EMPLOYMENT_DETAILS,
        component: EmploymentDetails,
        stage: StageName.HomeVisit,
        bothApplicants: true
    },
    {
        path: LANGUAGES_SPOKEN_IN_YOUR_HOME,
        component: LanguagesSpokenInYourHome,
        stage: StageName.HomeVisit
    },
    {
        path: PARTNERSHIP_STATUS,
        component: AreYouMarried,
        stage: StageName.HomeVisit,
        jointApplicantOnly: true
    },
    {
        path: DATE_OF_MARRIAGE_OR_PARTNERSHIP,
        component: MarriageDate,
        stage: StageName.HomeVisit,
        jointApplicantOnly: true
    },
    {
        path: DATE_OF_MOVE_IN_TOGETHER,
        component: MovedInTogetherDate,
        stage: StageName.HomeVisit,
        jointApplicantOnly: true
    },
    {
        path: FOSTERING_HISTORY,
        component: HaveYouPreviouslyApplied,
        stage: StageName.HomeVisit,
        bothApplicants: true
    },
    {
        path: ABOUT_YOUR_HEALTH,
        component: YourHealth,
        stage: StageName.HomeVisit,
        bothApplicants: true
    },
    {
        path: YOUR_INTEREST_IN_FOSTERING,
        component: AboutYourInterest,
        stage: StageName.HomeVisit
    },
    {
        path: TYPES_OF_FOSTERING,
        component: TypesOfFostering,
        stage: StageName.HomeVisit
    },
    {
        path: YOUR_HOUSEHOLD,
        component: AnyPeopleInYourHousehold,
        stage: StageName.HomeVisit
    },
    {
        path: WHO_LIVES_IN_YOUR_HOME,
        component: PeopleInYourHousehold,
        stage: StageName.HomeVisit
    },
    {
        path: DO_YOU_HAVE_ANY_PETS,
        component: DoYouHaveAnyPets,
        stage: StageName.HomeVisit
    },
    {
        path: CHILDREN_UNDER_SIXTEEN_LIVING_AWAY,
        component: AnyUnderSixteens,
        stage: StageName.HomeVisit,
        bothApplicants: true
    },
    {
        path: ABOUT_CHILDREN_UNDER_SIXTEEN_LIVING_AWAY,
        component: AboutUnderSixteens,
        stage: StageName.HomeVisit,
        bothApplicants: true
    },
    {
        path: CHILDREN_OVER_SIXTEEN_LIVING_AWAY,
        component: AnyOverSixteens,
        stage: StageName.HomeVisit,
        bothApplicants: true
    },
    {
        path: ABOUT_CHILDREN_OVER_SIXTEEN_LIVING_AWAY,
        component: AboutOverSixteens,
        stage: StageName.HomeVisit,
        bothApplicants: true
    },
    {
        path: ABOUT_YOUR_GP,
        component: YourGpDetails,
        stage: StageName.Application,
        bothApplicants: true
    },
    {
        path: FAMILY_REFERENCE,
        component: FamilyReference,
        stage: StageName.Application
    },
    {
        path: FIRST_PERSONAL_REFERENCE,
        component: FirstPersonalReference,
        stage: StageName.Application
    },
    {
        path: SECOND_PERSONAL_REFERENCE,
        component: SecondPersonalReference,
        stage: StageName.Application
    },
    {
        path: RELATIONSHIP_TO_COUNCIL_EMPLOYEES,
        component: DoYouKnowCouncillors,
        stage: StageName.Application,
        bothApplicants: true
    }
]