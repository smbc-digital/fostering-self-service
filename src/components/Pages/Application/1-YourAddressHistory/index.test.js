import { React, mount, useContextMock, renderer } from 'helpers/SetupTest'
import { act } from 'react-dom/test-utils'
import YourAddressHistory from './index'
import { Applicant } from 'constants'
import * as helpers from 'helpers'
import { START_PAGE, ABOUT_YOUR_GP, ADDRESS_HISTORY } from 'routes'

describe('YourAddressHistory', () => {
	const onChangeStatusMock = jest.fn()
	const onChangeTargetMock = jest.fn()

	beforeEach(() => {
		useContextMock.mockReturnValue({
			currentApplicant: Applicant.FirstApplicant,
			onChangeStatus: onChangeStatusMock,
			onChangeTarget: onChangeTargetMock,
			statuses: {
				addressHistoryStatus: 0
			},
			firstApplicant: {
				firstName: {
					value: 'First',
					isValid: true
				},
				lastName: {
					value: 'Applicant',
					isValid: true
				},
				addressHistory: {
					value: [{
						address: {
							addressLine1: {
								value: 'addressLine1',
								isValid: true
							},
							addressLine2: {
								value: 'addressLine2',
								isValid: true
							},
							town: {
								value: 'town',
								isValid: true
							},
							county: {
								value: 'county',
								isValid: true
							},
							country: {
								value: 'country',
								isValid: true
							},
							postcode: {
								value: 'postcode',
								isValid: true
							}
						},
						dateFrom: {
							value: '01-01-2009',
							isValid: true
						}

					}],
					isValid: true
				}
			},
			secondApplicant: {
				firstName: {
					value: 'Second',
					isValid: true
				},
				lastName: {
					value: 'Applicant',
					isValid: true
				},
				addressHistory: {
					value: [{
						address: {
							addressLine1: {
								value: 'secondaddressLine1',
								isValid: true
							},
							addressLine2: {
								value: 'secondaddressLine2',
								isValid: true
							},
							town: {
								value: 'secondtown',
								isValid: true
							},
							county: {
								value: 'secondcounty',
								isValid: true
							},
							country: {
								value: 'secondcountry',
								isValid: true
							},
							postcode: {
								value: 'secondpostcode',
								isValid: true
							}
						},
						dateFrom: {
							value: '01-06-2007',
							isValid: true
						}

					}],
					isValid: true
				}
			}
		})
	})

	it('should push to next page on submit, when single applicant', async () => {
		// Arrange
		const history = {
			push: jest.fn()
		}

		const match = {
			params: undefined
		}

		useContextMock.mockReturnValue({
			currentApplicant: Applicant.FirstApplicant,
			onChangeStatus: onChangeStatusMock,
			onChangeTarget: onChangeTargetMock,
			statuses: {
				addressHistoryStatus: 0
			},
			firstApplicant: {
				addressHistory: {
					value: [{
						address: {
							addressLine1: {
								value: 'addressLine1',
								isValid: true
							},
							addressLine2: {
								value: 'addressLine2',
								isValid: true
							},
							town: {
								value: 'town',
								isValid: true
							},
							county: {
								value: 'county',
								isValid: true
							},
							country: {
								value: 'country',
								isValid: true
							},
							postcode: {
								value: 'postcode',
								isValid: true
							}
						},
						dateFrom: {
							value: '01-01-2009',
							isValid: true
						}

					}],
					isValid: true
				}
			}
		})

		helpers.updateApplicationForm = jest.fn().mockReturnValue(Promise.resolve(0))

		const wrapper = mount(<YourAddressHistory history={history} match={match}/>)

		// Act
		await act(async () => {
			await wrapper.find('form').simulate('submit')
			await Promise.resolve()
		})

		// Assert
		expect(history.push).toHaveBeenCalledWith(ABOUT_YOUR_GP)
	})

	it('should push to YourAddressHistory on submit, when on first applicant but joint application', async () => {
		// Arrange
		const history = {
			push: jest.fn()
		}

		const match = {
			params: undefined
		}

		helpers.updateApplicationForm = jest.fn().mockReturnValue(Promise.resolve(0))

		const wrapper = mount(<YourAddressHistory history={history} match={match}/>)

		// Act
		await act(async () => {
			await wrapper.find('form').simulate('submit')
			await Promise.resolve()
		})

		// Assert
		expect(history.push).toHaveBeenCalledWith(ADDRESS_HISTORY + '/second-applicant')
	})

	it('should push to GP Details on submit, when on second applicant', async () => {
		// Arrange
		const history = {
			push: jest.fn()
		}

		const match = {
			params: ['second-applicant']
		}

		helpers.updateApplicationForm = jest.fn().mockReturnValue(Promise.resolve(0))

		const wrapper = mount(<YourAddressHistory history={history} match={match}/>)

		// Act
		await act(async () => {
			await wrapper.find('form').simulate('submit')
			await Promise.resolve()
		})

		// Assert
		expect(history.push).toHaveBeenCalledWith(ABOUT_YOUR_GP)
	})

	it('should call updateApplicationForm on form submit', async () => {
		// Arrange
		const history = {
			push: jest.fn()
		}

		const match = {
			params: ['second-applicant']
		}

		helpers.updateApplicationForm = jest.fn().mockReturnValue(Promise.resolve(0))
		
		const wrapper = mount(<YourAddressHistory history={history} match={match}/>)

		// Act
		await act(async () => {
			await wrapper.find('form').simulate('submit')
			await Promise.resolve()
		})

		// Assert
		expect(helpers.updateApplicationForm).toHaveBeenCalled()
	})

	it('should call onChangeStatus on form submit', async () => {
		// Arrange
		const history = {
			push: jest.fn()
		}

		const match = {
			params: ['second-applicant']
		}

		helpers.updateApplicationForm = jest.fn().mockReturnValue(Promise.resolve(0))

		const wrapper = mount(<YourAddressHistory history={history} match={match}/>)

		// Act
		await act(async () => {
			await wrapper.find('form').simulate('submit')
			await Promise.resolve()
		})

		// Assert
		expect(onChangeStatusMock).toHaveBeenCalledWith('addressHistoryStatus', 0)
	})

	it('should call updateApplicationForm on save and go back click', async () => {
		// Arrange
		const history = {
			push: jest.fn()
		}

		const match = {
			params: undefined
		}

		helpers.updateApplicationForm = jest.fn().mockReturnValue(Promise.resolve(0))

		const wrapper = mount(<YourAddressHistory history={history} match={match}/>)

		// Act
		await act(async () => {
			await wrapper.find('button').at(1).simulate('click')
			await Promise.resolve()
		})

		// Assert
		expect(helpers.updateApplicationForm).toHaveBeenCalled()
	})

	it('should push to start page on save and go back click', async () => {
		// Arrange
		const history = {
			push: jest.fn()
		}

		const match = {
			params: ['second-applicant']
		}

		helpers.updateApplicationForm = jest.fn().mockReturnValue(Promise.resolve(0))

		const wrapper = mount(<YourAddressHistory history={history} match={match}/>)

		// Act
		await act(async () => {
			await wrapper.find('button').at(1).simulate('click')
			await Promise.resolve()
		})

		// Assert
		expect(history.push).toHaveBeenCalledWith(START_PAGE)
	})

	it('should push to error page on updateApplicationForm error', async () => {
		// Arrange
		const history = {
			push: jest.fn()
		}

		const match = {
			params: undefined
		}

		helpers.updateApplicationForm = jest.fn().mockImplementation(() => { throw new Error() })

		const wrapper = mount(<YourAddressHistory history={history} match={match}/>)

		// Act
		await act(async () => await wrapper.find('form').simulate('submit'))

		// Assert
		expect(history.push).toHaveBeenCalledWith('/error')
	})

	it('should set value when dateFrom input changed', () => {
		// Arrange
		const match = {
			params: undefined
		}

		const wrapper = mount(<YourAddressHistory history={{}} match={match}/>)

		// Act
		onChangeTargetMock.mockReset()
		wrapper.find('input[name="year"]').simulate('change', { target: {value: '', name: 'year'} })

		// Assert
		expect(onChangeTargetMock).toHaveBeenCalledWith({'target': {'name': 'addressHistory', 'value': [{'address': {'addressLine1': {'isValid': true, 'value': 'addressLine1'}, 'addressLine2': {'isValid': true, 'value': 'addressLine2'}, 'country': {'isValid': true, 'value': 'country'}, 'county': {'isValid': true, 'value': 'county'}, 'postcode': {'isValid': true, 'value': 'postcode'}, 'town': {'isValid': true, 'value': 'town'}}, 'dateFrom': {'isValid': false, 'value': null}}]}}, false, 'firstApplicant')
	})

	
	it('should set value when address input changed', () => {
		// Arrange
		useContextMock.mockReturnValue({
			currentApplicant: Applicant.FirstApplicant,
			onChangeStatus: onChangeStatusMock,
			onChangeTarget: onChangeTargetMock,
			country: [{value: 'test', name: 'test'}],
			statuses: {
				addressHistoryStatus: 0
			},
			firstApplicant: {
				addressHistory: {
					value: [{
						address: {
							addressLine1: {
								value: 'addressLine1',
								isValid: true
							},
							addressLine2: {
								value: 'addressLine2',
								isValid: true
							},
							town: {
								value: 'town',
								isValid: true
							},
							county: {
								value: 'county',
								isValid: true
							},
							country: {
								value: 'country',
								isValid: true
							},
							postcode: {
								value: 'postcode',
								isValid: true
							}
						},
						dateFrom: {
							value: '01-01-2009',
							isValid: true
						}

					},
					{
						address: {
							addressLine1: {
								value: 'addressLine1',
								isValid: true
							},
							addressLine2: {
								value: 'addressLine2',
								isValid: true
							},
							town: {
								value: 'town',
								isValid: true
							},
							county: {
								value: 'county',
								isValid: true
							},
							country: {
								value: 'country',
								isValid: true
							},
							postcode: {
								value: 'postcode',
								isValid: true
							}
						},
						dateFrom: {
							value: '01-01-2009',
							isValid: true
						}

					}],
					isValid: true
				}
			}
		})

		const match = {
			params: undefined
		}

		const wrapper = mount(<YourAddressHistory history={{}} match={match}/>)

		// Act
		onChangeTargetMock.mockReset()
		wrapper.find('input[name="addressLine1"]').simulate('change', { target: {value: 'new address line 1 value', name: 'addressLine1'} })

		// Assert
		expect(onChangeTargetMock).toHaveBeenCalledWith({'target': {'name': 'addressHistory', 'value': [{'address': {'addressLine1': {'isValid': true, 'value': 'addressLine1'}, 'addressLine2': {'isValid': true, 'value': 'addressLine2'}, 'country': {'isValid': true, 'value': 'country'}, 'county': {'isValid': true, 'value': 'county'}, 'postcode': {'isValid': true, 'value': 'postcode'}, 'town': {'isValid': true, 'value': 'town'}}, 'dateFrom': {'isValid': true,
'value': '01-01-2009'}}, {'address': {'addressLine1': {'isValid': true, 'value': 'new address line 1 value'}, 'addressLine2': {'isValid': true, 'value': 'addressLine2'}, 'country': {'isValid': true, 'value': 'country'}, 'county': {'isValid': true, 'value': 'county'}, 'postcode': {'isValid': true, 'value': 'postcode'}, 'town': {'isValid': true, 'value': 'town'}}, 'dateFrom': {'isValid': true, 'value': '01-01-2009'}}]}}, true, 'firstApplicant')
	})

	it('should checkIsValid when multiple addresses', () => {
		// Arrange
		useContextMock.mockReturnValue({
			currentApplicant: Applicant.FirstApplicant,
			onChangeStatus: onChangeStatusMock,
			onChangeTarget: onChangeTargetMock,
			country: [{value: 'test', name: 'test'}],
			statuses: {
				addressHistoryStatus: 0
			},
			firstApplicant: {
				addressHistory: {
					value: [{
						address: {
							addressLine1: {
								value: 'addressLine1',
								isValid: true
							},
							addressLine2: {
								value: 'addressLine2',
								isValid: true
							},
							town: {
								value: 'town',
								isValid: true
							},
							county: {
								value: 'county',
								isValid: true
							},
							country: {
								value: 'country',
								isValid: true
							},
							postcode: {
								value: 'postcode',
								isValid: true
							}
						},
						dateFrom: {
							value: '01-01-2009',
							isValid: true
						}

					},
					{
						address: {
							addressLine1: {
								value: 'addressLine1',
								isValid: true
							},
							addressLine2: {
								value: 'addressLine2',
								isValid: true
							},
							town: {
								value: 'town',
								isValid: true
							},
							county: {
								value: 'county',
								isValid: true
							},
							country: {
								value: 'country',
								isValid: true
							},
							postcode: {
								value: 'postcode',
								isValid: true
							}
						},
						dateFrom: {
							value: '01-01-2009',
							isValid: true
						}

					}],
					isValid: true
				}
			}
		})

		const match = {
			params: undefined
		}

		const wrapper = mount(<YourAddressHistory history={{}} match={match}/>)

		// Assert
		expect(wrapper.find('Button').first().prop('disabled')).toEqual(undefined)

	})

	describe('snapshot', () => {
		it('renders firstApplicant correctly for single applicant', () => {
			useContextMock.mockReturnValue({
				currentApplicant: Applicant.FirstApplicant,
				onChangeStatus: onChangeStatusMock,
				onChangeTarget: onChangeTargetMock,
				country: [{value: 'test', name: 'test'}],
				statuses: {
					addressHistoryStatus: 0
				},
				firstApplicant: {
					addressHistory: {
						value: [{
							address: {
								addressLine1: {
									value: 'addressLine1',
									isValid: true
								},
								addressLine2: {
									value: 'addressLine2',
									isValid: true
								},
								town: {
									value: 'town',
									isValid: true
								},
								county: {
									value: 'county',
									isValid: true
								},
								country: {
									value: 'test',
									isValid: true
								},
								postcode: {
									value: 'postcode',
									isValid: true
								}
							},
							dateFrom: undefined
						}],
						isValid: false
					}
				}
			})
			
			const match = {
				params: undefined
			}

			const tree = renderer
				.create(<YourAddressHistory history={{}} match={match}/>)
				.toJSON()

			expect(tree).toMatchSnapshot()
		})

		it('renders firstApplicant correctly for single applicant when first date entered is not more than 10 years ago', () => {
			useContextMock.mockReturnValue({
				currentApplicant: Applicant.FirstApplicant,
				onChangeStatus: onChangeStatusMock,
				onChangeTarget: onChangeTargetMock,
				country: [{value: 'test', name: 'test'}],
				statuses: {
					addressHistoryStatus: 0
				},
				firstApplicant: {
					addressHistory: {
						value: [{
							address: {
								addressLine1: {
									value: 'addressLine1',
									isValid: true
								},
								addressLine2: {
									value: 'addressLine2',
									isValid: true
								},
								town: {
									value: 'town',
									isValid: true
								},
								county: {
									value: 'county',
									isValid: true
								},
								country: {
									value: 'test',
									isValid: true
								},
								postcode: {
									value: 'postcode',
									isValid: true
								}
							},
							dateFrom: {
								value: '01-01-2015',
								isValid: true
							}
						}],
						isValid: false
					}
				}
			})
			
			const match = {
				params: undefined
			}

			const tree = renderer
				.create(<YourAddressHistory history={{}} match={match}/>)
				.toJSON()

			expect(tree).toMatchSnapshot()
		})

		it('renders second applicant correctly when no data', () => {
			useContextMock.mockReturnValue({
				currentApplicant: Applicant.FirstApplicant,
				onChangeStatus: onChangeStatusMock,
				onChangeTarget: onChangeTargetMock,
				country: [{value: 'test', name: 'test'}],
				statuses: {
					addressHistoryStatus: 0
				},
				firstApplicant: {
					firstName: {
						value: 'First',
						isValid: true
					},
					lastName: {
						value: 'Applicant',
						isValid: true
					},
					addressHistory: {
						value: [{
							address: {},
							dateFrom: {}
						}],
						isValid: false
					}
				},
				secondApplicant: {
					firstName: {
						value: 'Second',
						isValid: true
					},
					lastName: {
						value: 'Applicant',
						isValid: true
					},
					addressHistory: {
						value: [{
							address: {},
							dateFrom: {
								value: '02-06-2016',
								isValid: true
							}
						}, {
							address: undefined,
							dateFrom: undefined
						}],
						isValid: false
					}
				}
			})

			const match = {
				params: ['second-applicant']
			}

			const tree = renderer
				.create(<YourAddressHistory history={{}} match={match}/>)
				.toJSON()
			
			expect(tree).toMatchSnapshot()
		})
	})
})