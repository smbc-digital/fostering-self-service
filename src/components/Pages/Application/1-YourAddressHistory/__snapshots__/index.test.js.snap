// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YourAddressHistory snapshot renders firstApplicant correctly for single applicant 1`] = `
Array [
  <h1>
    Your fostering journey
  </h1>,
  <h2>
    Your address history
  </h2>,
  <form
    onSubmit={[Function]}
  >
    <h3
      className="no-margin-bottom"
    >
      Tell us when you moved into your current address
    </h3>
    <fieldset
      className="input-memorable-date"
    >
      <p>
        For example, 3 1980
      </p>
      <div>
        <label>
          Month
        </label>
        <input
          aria-label="month"
          className=""
          name="month"
          onChange={[Function]}
          onKeyDown={[Function]}
          type="number"
          value=""
        />
      </div>
      <div>
        <label>
          Year
        </label>
        <input
          aria-label="year"
          className=""
          name="year"
          onChange={[Function]}
          onKeyDown={[Function]}
          type="number"
          value=""
        />
      </div>
    </fieldset>
    <button
      className="button-primary"
      disabled={true}
      id={undefined}
      onClick={undefined}
      type="submit"
    >
      Save and next step
    </button>
    <p>
      <a
        href="#"
        onClick={[Function]}
      >
        Previous
      </a>
    </p>
    <button
      className="button-inverted left-chevron"
      disabled={true}
      id={undefined}
      onClick={[Function]}
      type="submit"
    >
      Save and go back to fostering area
    </button>
  </form>,
]
`;

exports[`YourAddressHistory snapshot renders firstApplicant correctly for single applicant when first date entered is not more than 10 years ago 1`] = `
Array [
  <h1>
    Your fostering journey
  </h1>,
  <h2>
    Your address history
  </h2>,
  <div>
    <section
      className="alert-information-form"
    >
      <span
        className="fa fa-info-circle"
      />
      <p
        dangerouslySetInnerHTML={
          Object {
            "__html": "You must tell us where you’ve lived since August 2009, starting with the most recent address and working backwards.",
          }
        }
      />
    </section>
  </div>,
  <form
    onSubmit={[Function]}
  >
    <h3
      className="no-margin-bottom"
    >
      Tell us when you moved into your current address
    </h3>
    <fieldset
      className="input-memorable-date"
    >
      <p>
        For example, 3 1980
      </p>
      <div>
        <label>
          Month
        </label>
        <input
          aria-label="month"
          className=""
          name="month"
          onChange={[Function]}
          onKeyDown={[Function]}
          type="number"
          value="1"
        />
      </div>
      <div>
        <label>
          Year
        </label>
        <input
          aria-label="year"
          className=""
          name="year"
          onChange={[Function]}
          onKeyDown={[Function]}
          type="number"
          value="2015"
        />
      </div>
    </fieldset>
    <a
      className="anchor-with-icon"
      href="#"
      onClick={[Function]}
    >
       
      <i
        className="fa fa-plus-square fa-lg"
      />
      Add another address
    </a>
    <button
      className="button-primary"
      disabled={true}
      id={undefined}
      onClick={undefined}
      type="submit"
    >
      Save and next step
    </button>
    <p>
      <a
        href="#"
        onClick={[Function]}
      >
        Previous
      </a>
    </p>
    <button
      className="button-inverted left-chevron"
      disabled={true}
      id={undefined}
      onClick={[Function]}
      type="submit"
    >
      Save and go back to fostering area
    </button>
  </form>,
]
`;

exports[`YourAddressHistory snapshot renders second applicant correctly when no data 1`] = `
Array [
  <h1>
    Your fostering journey
  </h1>,
  <h2>
    Your address history
  </h2>,
  <div>
    <section
      className="alert-information-form"
    >
      <span
        className="fa fa-info-circle"
      />
      <p
        dangerouslySetInnerHTML={
          Object {
            "__html": "You must tell us where you’ve lived since August 2009, starting with the most recent address and working backwards.",
          }
        }
      />
    </section>
  </div>,
  <p
    className="h3"
  >
    Second
     
    Applicant
  </p>,
  <form
    onSubmit={[Function]}
  >
    <h3
      className="no-margin-bottom"
    >
      Tell us when you moved into your current address
    </h3>
    <fieldset
      className="input-memorable-date"
    >
      <p>
        For example, 3 1980
      </p>
      <div>
        <label>
          Month
        </label>
        <input
          aria-label="month"
          className=""
          name="month"
          onChange={[Function]}
          onKeyDown={[Function]}
          type="number"
          value="6"
        />
      </div>
      <div>
        <label>
          Year
        </label>
        <input
          aria-label="year"
          className=""
          name="year"
          onChange={[Function]}
          onKeyDown={[Function]}
          type="number"
          value="2016"
        />
      </div>
    </fieldset>
    <hr />
    <h3>
      Tell us your previous address
    </h3>
    <label
      className={undefined}
      htmlFor="addressLine1"
    >
      Address line 1
    </label>
    <input
      aria-label="Address line 1"
      aria-live="off"
      className=""
      id="addressLine1"
      inputMode={undefined}
      maxLength={60}
      min={undefined}
      name="addressLine1"
      onChange={[Function]}
      onKeyDown={[Function]}
      pattern={undefined}
      type="text"
      value=""
    />
    <h3
      aria-hidden="true"
      className="optional"
    >
      Address line 2
    </h3>
    <p
      aria-hidden={true}
      className="optional"
    >
      (optional)
    </p>
    <label
      className="optional sr-only"
      htmlFor="addressLine2"
    >
      Address line 2 optional
    </label>
    <input
      aria-label=""
      aria-live=""
      className=""
      id="addressLine2"
      inputMode={undefined}
      maxLength={60}
      min={undefined}
      name="addressLine2"
      onChange={[Function]}
      onKeyDown={[Function]}
      pattern={undefined}
      type="text"
      value=""
    />
    <label
      className={undefined}
      htmlFor="town"
    >
      City or town
    </label>
    <input
      aria-label="City or town"
      aria-live="off"
      className=""
      id="town"
      inputMode={undefined}
      maxLength={60}
      min={undefined}
      name="town"
      onChange={[Function]}
      onKeyDown={[Function]}
      pattern={undefined}
      type="text"
      value=""
    />
    <h3
      aria-hidden="true"
      className="optional"
    >
      County or province
    </h3>
    <p
      aria-hidden={true}
      className="optional"
    >
      (optional)
    </p>
    <label
      className="optional sr-only"
      htmlFor="county"
    >
      County or province optional
    </label>
    <input
      aria-label=""
      aria-live=""
      className=""
      id="county"
      inputMode={undefined}
      maxLength={60}
      min={undefined}
      name="county"
      onChange={[Function]}
      onKeyDown={[Function]}
      pattern={undefined}
      type="text"
      value=""
    />
    <fieldset>
      <label
        className=""
        htmlFor="country"
      >
        Country
      </label>
      <span
        aria-controls="country"
        aria-expanded={false}
        aria-labelledby="country"
        className="sr-only"
        role="combobox"
      >
        country
      </span>
      <select
        aria-live="off"
        id="country"
        name="country"
        onChange={[Function]}
        value=""
      >
        <option
          disabled={true}
          value=""
        >
          Select a country...
        </option>
        <option
          value="test"
        >
          test
        </option>
      </select>
    </fieldset>
    <label
      className="optional"
      htmlFor="postcode"
    >
      Postcode
    </label>
    <input
      aria-label=""
      aria-live=""
      className=""
      id="postcode"
      inputMode={undefined}
      maxLength={60}
      min={undefined}
      name="postcode"
      onChange={[Function]}
      onKeyDown={[Function]}
      pattern={undefined}
      type="text"
      value=""
    />
    <h3
      className="no-margin-bottom"
    >
      Tell us when you moved into this address
    </h3>
    <fieldset
      className="input-memorable-date"
    >
      <p>
        For example, 3 1980
      </p>
      <div>
        <label>
          Month
        </label>
        <input
          aria-label="month"
          className=""
          name="month"
          onChange={[Function]}
          onKeyDown={[Function]}
          type="number"
          value=""
        />
      </div>
      <div>
        <label>
          Year
        </label>
        <input
          aria-label="year"
          className=""
          name="year"
          onChange={[Function]}
          onKeyDown={[Function]}
          type="number"
          value=""
        />
      </div>
    </fieldset>
    <a
      className="anchor-with-icon"
      href="#"
      onClick={[Function]}
    >
       
      <i
        className="fa fa-minus-square fa-lg"
      />
      Remove this address
    </a>
    <a
      className="anchor-with-icon"
      href="#"
      onClick={[Function]}
    >
       
      <i
        className="fa fa-plus-square fa-lg"
      />
      Add another address
    </a>
    <button
      className="button-primary"
      disabled={true}
      id={undefined}
      onClick={undefined}
      type="submit"
    >
      Save and next step
    </button>
    <p>
      <a
        href="#"
        onClick={[Function]}
      >
        Previous
      </a>
    </p>
    <button
      className="button-inverted left-chevron"
      disabled={true}
      id={undefined}
      onClick={[Function]}
      type="submit"
    >
      Save and go back to fostering area
    </button>
  </form>,
]
`;
